src/OpenLoco/GameCommands/Cheat.cpp

```diff
@@ -33,13 +33,21 @@ namespace OpenLoco::GameCommands
                 auto* roadElement = element.as<RoadElement>();
                 if (roadElement != nullptr)
                 {
+                    // check to verify that roadElement is owned by the target company
+                    if (roadElement->owner() != targetCompanyId)
+                        continue;
+
                     roadElement->setOwner(ourCompanyId);
                     continue;
```

Comment:
I think we should invert the logic here. A tile can contain several road elements (e.g. crossings/turns) and indeed both roads and tracks at the same time (e.g. trams). Invoking `continue` on line 38 might miss some of them. I think my original code suffers the same issue.

Therefore, I suggest we simplify this block to:
```suggestion
                    if (roadElement->owner() == targetCompanyId)
                        roadElement->setOwner(ourCompanyId);
```

With a similar change below. Thoughts?

Reply:
wouldn't we still need continue? doesn't the continue step through the for loop?

Reply:
No, continue is only needed if the normal control flow needs to be interrupted. At the end of the block, the control flow automatically loops back to the beginning of the loop.

Reply:
ok this is done

Reply:
Ahh, no, I realise I was misreading the context. You were right, we do need to invoke continue manually here to not reinterpret the same element twice.

Reply:
does continue need to be inside or outside of the if statement? or does it matter?

Reply:
I have pushed the necessary fix. No worries.